support/types.go

```diff
@@ -0,0 +1,100 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package support
+
+import (
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/e2e-framework/klient"
+)
+
+type ClusterOpts func(c E2EClusterProvider)
+
+type E2EClusterProvider interface {
+	// WithName is used to configure the cluster Name that should be used while setting up the cluster. Might
+	// Not apply for all providers.
+	WithName(name string) E2EClusterProvider
+
+	// WithVersion helps you override the default version used while using the cluster provider.
+	// This can be useful in providing a mechanism to the end users where they want to test their
+	// code against a certain specific version of k8s that is not the default one configured
+	// for the provider
+	WithVersion(version string) E2EClusterProvider
+
+	// WithPath heps you customize the executable binary that is used to back the cluster provider.
+	// This is useful in cases where your binary is present in a non standard location output of the
+	// PATH variable and you want to use that instead of framework trying to install one on it's own.
+	WithPath(path string) E2EClusterProvider
+
+	// WithOpts provides a way to customize the options that can be used while setting up the
+	// cluster using the providers such as kind or kwok or anything else. These helpers can be
+	// leveraged to setup arguments or configuration values that can be provided while performing
+	// the cluster bring up
+	WithOpts(opts ...ClusterOpts) E2EClusterProvider
+
+	// Create Provides an interface to start the cluster creation workflow using the selected provider
+	Create(args ...string) (string, error)
+
+	// CreateWithConfig is used to provide a mechanism where cluster providers that take an input config
+	// file and then setup the cluster accordingly. This can be used to provide input such as kind config
+	CreateWithConfig(configFile string) (string, error)
+
+	// GetKubeconfig provides a way to extract the kubeconfig file associated with the cluster in question
+	// using the cluster provider native way
+	GetKubeconfig() string
+
+	// GetKubeCtlContext is used to extract the kubectl context to be used while performing the operation
+	GetKubeCtlContext() string
+
+	// ExportLogs is used to export the cluster logs via the cluster provider native workflow. This
+	// can be used to export logs from the cluster after test failures for example to analyze the test
+	// failures better after the fact.
+	ExportLogs(dest string) error
+
+	// Destroy is used to cleanup a cluster brought up as part of the test workflow
+	Destroy() error
+
+	// LoadDockerImage is used to load a set of Docker images to the cluster via the cluster provider native workflow
+	// Not every provider will have a mechanism like this/need to do this. So, providers that do not have this support
+	// can just provide a no-op implementation to be compliant with the interface
+	LoadDockerImage(image string) error
```

Comment:
```suggestion
	LoadImage(image string) error
```

Images may not just be for Docker.